### DISCUSSION TITLE
How to increase the specificity of tailwind util classes

### DISCUSSION THREAD

**Discussion by jithureddy** [tailwindlabs/tailwindcss#5080] [Help]
I have got the cms application with bootstrap, and embedding react application with tailwind css into it.
There are clashes with both css framework util classes. 

I enabled !important via tailwind configuration then have the below issue, Also tried with selector strategy via config as `important: '.tailwind-app'`, but still bootstrap !important rules override. I need to increase the specificity and add add !important to tailwind classes so that application will work without affected. 

I don't want to use pre-fixing styles for tailwind because, there are some components loaded from my component library, in which I don't want to make changes

![image](https://user-images.githubusercontent.com/5911189/126963145-03fcd618-d4bf-463e-aabd-bab6e597378a.png)

**Comment by jithureddy** [comment]
@adamwathan Please suggest a way to override some rules with !important using tailwind configuration

**Comment by simonswiss** [comment]
Hey!

If both `py-4` classes (Bootstrap, Tailwind) have an `!important` flag, the class that is defined _later_ in your CSS bundle will take precedence. That's how the cascade works.

From your screenshot above, I can dedutc that your `_spacing.scss` partial is generating the CSS _after_ the `main.css`, where your Tailwind classes are being generated?

A battle of `!important` is never a great starting point, but your best bet here is to make sure your Tailwind classes are generated _last_, after Bootstrap, in your CSS pipeline.

That way, your Tailwind `!important` will override the Boostrap `!important` 😅 

That will work for most "flat" selectors in Bootstrap (like `py-4, `text-center`, etc). If the classes are more specific (like `.container p`) and also `!important`, you'll be out of luck (unless you bring inline style weaponry to the party).

Hope it helps!

**Comment by seiyab** [comment]
I want to increase [specificity](https://developer.mozilla.org/en-US/docs/Web/CSS/Guides/Cascade/Specificity).
I'm not going to battle `!important`, just want specificity. Say, I want to override `border-slate-200` by `border-rose-500`.
What I want is something like:
```html
<!-- input -->
<div class="border-slate-200 border-rose-500^2" />
```
```css
/* output */
.borde-slate-200 {
  ...
}

.border-rose-500\^2.border-rose-500\^2 {
  ...
}
```

**Reply by wongjn** [reply]
You can double-up the class specificity via `[&&]:` variant: https://play.tailwindcss.com/o2lBBJjE6B

**Reply by seiyab** [reply]
@wongjn Thank you! It looks what I want!

**Reply by rizqinrr** [reply]
Tailwind actually has a built-in feature to achieve this exact behavior using **Arbitrary Variants**. You can use `[&&]:` to duplicate the selector and bump up its specificity without relying on `!important`.

Here is how you can write it:

```html
<div class="border-slate-200 [&&]:border-rose-500"></div>

```

When compiled, Tailwind will generate the exact `.class.class` stacking you were looking for:

```css
/* output */
.border-slate-200 {
  --tw-border-opacity: 1;
  border-color: rgb(226 232 240 / var(--tw-border-opacity));
}

.\[\&\&\]\:border-rose-500.\[\&\&\]\:border-rose-500 {
  --tw-border-opacity: 1;
  border-color: rgb(244 63 94 / var(--tw-border-opacity));
}

```

This perfectly overrides the `border-slate-200` by doubling the specificity of the rose border
